### PR TITLE
Unify network api

### PR DESCRIPTION
**Pull request by erkanerol**
Towards https://github.com/giantswarm/giantswarm/issues/26801

Applies the standard interface for network parameters.

### Testing

- [x] fresh install works
- [x] upgrade from previous version works

### Checklist

- [x] Update changelog in CHANGELOG.md.
- [x] Make sure `values.yaml` and `values.schema.json` are valid.
- [x] Update `/examples` if required.

### CODE REVIEW

File: helm/cluster-cloud-director/templates/cloud-provider-cloud-director-helmrelease.yaml [hunk 1]
```diff
@@ -49,7 +49,7 @@ spec:
         ovdcNetwork: {{ .Values.cloudDirector.ovdcNetwork | quote }}
         site: {{ .Values.cloudDirector.site | quote }}
         vAppName: {{ include "resource.default.name" $  | quote }}
-        vipSubnet: {{ .Values.network.loadBalancer.vipSubnet | quote }}
+        vipSubnet: {{ index .Values.connectivity.network.loadBalancers.cidrBlocks 0 | quote }}
```
Comment (kopiczko) [inline helm/cluster-cloud-director/templates/cloud-provider-cloud-director-helmrelease.yaml:52]: We allow multiple cidrBlocks here but only the first one is used here? What's the purpose?

Comment (erkanerol) [reply helm/cluster-cloud-director/templates/cloud-provider-cloud-director-helmrelease.yaml:52]: I just followed the same logic we use in other places. I am not sure whether it is possible to use dual-stack for loadbalancers too. @vxav is it possible?

Comment (vxav) [reply helm/cluster-cloud-director/templates/cloud-provider-cloud-director-helmrelease.yaml:52]: While I am all for unifying across providers, personally I am not a big fan of obfuscating those provider specific fields as they will not match the fields in the CRs and will make debugging, editing and such more complicated (I can see myself wondering what `cidrBlocks` is in a few months time). 

In terms of using an array, I don't think it is supported according to https://github.com/vmware/cluster-api-provider-cloud-director/blob/main/config/crd/bases/infrastructure.cluster.x-k8s.io_vcdclusters.yaml#L511

Comment (erkanerol) [reply helm/cluster-cloud-director/templates/cloud-provider-cloud-director-helmrelease.yaml]: I used `vipSubnet` again and it is a string now.

**Review by kopiczko — APPROVED** [review]

**Pull request merged**